### ISSUE TITLE
VMware VM Fails to load with Terraform

### ISSUE THREAD

**Issue by gyalowitzdtlr** [flatcar/Flatcar#272]
Since the update our terraform kubernetes spinups have all been failing to load.  They get stuck with that error start job /dev/dis error.  What is strange is that this issue just cropped up since they were working without issue before.  Could this all be caused by the issue with the guest OS that have cropped up?

**Comment by pothos** [comment]
I can't really follow what your setup looks like, but are you deploying new instances and they are stuck in the first boot? This can be an Ignition config issue if it's during the initramfs. Do you rely on fetching remote files in Ignition? This only works for DHCP, not static IP addresses.
Are you using the latest Stable images?

By the way, currently the VMware tools on the OEM partition are not updated and in case you rely on them, you have to copy them over. Tthere is another issue about that were a workaround was posted: https://github.com/kinvolk/Flatcar/issues/21#issuecomment-724205539

**Comment by gyalowitzdtlr** [comment]
Thank you for the response.  It is actually starting to look like it could be a few things.  I moved the flatcar image we are using to alpha and now it seems like the ignition userdata is not being used at all.  The ignition file is base64 encoded and supplied through terraform ignition config.  SO I am basically trying to figure out if its:
1.  Vmware tools
2.  L1TF and MDS mitigations we have in place in Vsphere.
3.  Terraform provider is now broken.  

Reading through these issues though it does seem like the L1TF and MDS kernal lockdowns more or less break guestinfo.  Basically the terraform modules that spun up this piece of infrastructure have been completely at rest and worked without issue.  The only major change was upgrading VMware to 7.0.1 and implement SCAV2.

**Comment by gyalowitzdtlr** [comment]
![image](https://user-images.githubusercontent.com/59739698/99865945-4b589380-2b7b-11eb-8baf-416e2762bd48.png)

![image](https://user-images.githubusercontent.com/59739698/99866023-f49f8980-2b7b-11eb-95a5-f03c20e288af.png)

![image](https://user-images.githubusercontent.com/59739698/99866347-9e801580-2b7e-11eb-905e-01c5220f0b75.png)

![image](https://user-images.githubusercontent.com/59739698/99866031-0123e200-2b7c-11eb-9a0d-872fae67a93f.png)

![image](https://user-images.githubusercontent.com/59739698/99866041-0d0fa400-2b7c-11eb-9d87-732a2b6a5a43.png)
![image](https://user-images.githubusercontent.com/59739698/99866080-59f37a80-2b7c-11eb-9281-8143c1078f67.png)
![image](https://user-images.githubusercontent.com/59739698/99866084-5fe95b80-2b7c-11eb-8a71-7303a77fcd1f.png)

![image](https://user-images.githubusercontent.com/59739698/99866104-8a3b1900-2b7c-11eb-84e8-e29d727437b4.png)
![image](https://user-images.githubusercontent.com/59739698/99866116-a8a11480-2b7c-11eb-8e57-6e0044978431.png)
![image](https://user-images.githubusercontent.com/59739698/99866190-177e6d80-2b7d-11eb-9885-c4861400fe7e.png)
![image](https://user-images.githubusercontent.com/59739698/99866194-1c432180-2b7d-11eb-840f-d715676bab8c.png)
![image](https://user-images.githubusercontent.com/59739698/99866237-7cd25e80-2b7d-11eb-8d49-ece2647553e5.png)

![image](https://user-images.githubusercontent.com/59739698/99866318-52cd6c00-2b7e-11eb-805f-5573375045a5.png)

![image](https://user-images.githubusercontent.com/59739698/99866412-fa4a9e80-2b7e-11eb-863f-eceabcc69187.png)

![image](https://user-images.githubusercontent.com/59739698/99866480-78a74080-2b7f-11eb-9215-5a3e89fa386e.png)

**Comment by pothos** [comment]
Thanks for the infos. Do you also have `ignition.config.data.encoding` set to `base64`? Do you need to pass the `guestinfo.` prefix or not in Terraform? Can you try out please if `/usr/share/oem/bin/vmware-rpctool 'info-get guestinfo.ignition.config.data'` works?

**Comment by gyalowitzdtlr** [comment]
The info-get result:
![image](https://user-images.githubusercontent.com/59739698/99997120-b4622600-2d8a-11eb-9107-681b73433139.png)

Encoding being passed in:
![image](https://user-images.githubusercontent.com/59739698/99997182-cc39aa00-2d8a-11eb-8dd3-414a2ffab75f.png)

How ignition is being passed in:
![image](https://user-images.githubusercontent.com/59739698/99997259-ea070f00-2d8a-11eb-91b5-5e9d046154fe.png)

Basically the ignition provider creates a rendered ignition file into json.  We then base64 encode it and pass it into the vmware we are spinning up as a vapp property that you see in the bottom picture.  Again, what is absolutely confusing is that this was working up until recently.  Is it possible that the intel vulnerability mitigations could be at the root of this?

**Comment by gyalowitzdtlr** [comment]
Here is a link to the KB article that I followed.
https://kb.vmware.com/s/article/55806

**Comment by gyalowitzdtlr** [comment]
Also, looking at my ignition config it looks like it is being generated using version 2.1.0 and I wonder if the new flatcar images or vmware upgrades is now no longer compatible with that version.  Would you know if it is?

**Comment by gyalowitzdtlr** [comment]
Also, using the following command:
![image](https://user-images.githubusercontent.com/59739698/100012049-a6b79b00-2da0-11eb-9fc8-ad38b61ba9ad.png)

You can see the base64 encoded guestinfo variable being read:
![image](https://user-images.githubusercontent.com/59739698/100012079-b46d2080-2da0-11eb-8590-47a47301cef5.png)

**Comment by gyalowitzdtlr** [comment]
Good news is that I managed to get ignition to load again.  The terraform provider for ignition is pulling from the fork that only does 3.x spec which is a problem for Flatcar, at least from what I read.  I also had to go in and seriously tweak the versions of providers that terraform is pulling to make sure it's the correct version and uniform across all modules.  Now the only issue is the vmwaretools service failing.
